tests/cupy_tests/math_tests/test_matmul.py

```diff
@@ -39,30 +39,40 @@ class TestMatmul(unittest.TestCase):
 
     # _multiprocess_can_split_ = True
 
-    def setUp(self):
-        self.x1 = numpy.random.randn(*self.shape_pair[0])
-        self.x2 = numpy.random.randn(*self.shape_pair[1])
-
     @unittest.skipUnless(sys.version_info >= (3, 5),
                          'Only for Python3.5 or higher')
     @testing.with_requires('numpy>=1.10')
-    @testing.for_all_dtypes(name='dtype1')
-    @testing.for_all_dtypes(name='dtype2')
-    @testing.numpy_cupy_allclose(rtol=1e-4, atol=1e-3)  # required for uint8
+    @testing.for_all_dtypes(name='dtype1', no_float16=True)
+    @testing.for_all_dtypes(name='dtype2', no_float16=True)
+    @testing.numpy_cupy_allclose(rtol=1e-3, atol=1e-3)  # required for uint8
     def test_operator_matmul(self, xp, dtype1, dtype2):
-        if not numpy.result_type(dtype1, dtype2) == numpy.float32:
-            return xp.array([])
-        x1 = xp.array(self.x1, dtype=dtype1)
-        x2 = xp.array(self.x2, dtype=dtype2)
+        x1 = testing.shaped_arange(self.shape_pair[0], xp, dtype1)
+        x2 = testing.shaped_arange(self.shape_pair[1], xp, dtype2)
         return operator.matmul(x1, x2)
 
+    # Since calculation accuracy is bad for (float16, uint8).
```

Comment:
Do you mean the calculation accuracy is bad only between float16 and uint8? If so, how about testing the cases between float16 and non-uint8 types and between uint8 and non-float16 types? And is it still possible to include low-accuracy cases with low tolerance parameters?